src/world/lua.rs

```diff
@@ -129,6 +129,72 @@ impl LuaPlayer {
         self.create_segment_self(op_code, data);
     }
 
+    fn unlock_action(&mut self, id: u32) {
+        let op_code = ServerZoneIpcType::ActorControlSelf;
+        let data = ServerZoneIpcData::ActorControlSelf(ActorControlSelf {
+            category: ActorControlCategory::ToggleActionUnlock { id, unlocked: true },
+        });
+
+        self.create_segment_self(op_code, data);
+    }
+
+    fn set_speed(&mut self, speed: u16) {
+        let op_code = ServerZoneIpcType::ActorControlSelf;
+        let data = ServerZoneIpcData::ActorControlSelf(ActorControlSelf {
+            category: ActorControlCategory::Flee { speed },
+        });
+
+        self.create_segment_self(op_code, data);
+    }
+
+    fn toggle_wireframe(&mut self) {
+        let op_code = ServerZoneIpcType::ActorControlSelf;
+        let data = ServerZoneIpcData::ActorControlSelf(ActorControlSelf {
+            category: ActorControlCategory::ToggleWireframeRendering(),
+        });
+
+        self.create_segment_self(op_code, data);
+    }
+
+    fn toggle_invisiblity(&mut self) {
+        let op_code = ServerZoneIpcType::ActorControlSelf;
+        /* TODO: This doesn't actually act as a toggle right now, because we can't update
+         * the player's gm_invisible state from this side it seems. How should this be resolved..? */
+        self.player_data.gm_invisible = !self.player_data.gm_invisible;
```

Comment:
Right, the reason why this doesn't work is because Lua is given a _copy_ of PlayerData (this is not obvious, sorry for not documenting that :))

The two ways to go about it is:
* Create a Task that is processed to toggle invisibility, probably overkill here. Tasks are handled in `Connection::process_lua_player`.
* Copy the `gm_invisible` status back somewhere in lines 800-815 in `kawari-world.rs`

Reply:
I opted for the first way, copying it directly after process_lua_player felt hacky to me.